### DISCUSSION TITLE
Sort label option for GENERIC_CHART_AXES

### DISCUSSION THREAD

**Discussion by kamalkeshavani-aiinside** [apache/superset#20885] [Ideas]
Legacy Bar chart sorted the bars by their value, and had an option to sort them by X-axis labels.

With new 'GENERIC_CHART_AXES' flag, we can use labels on echarts X-axis, but they are always sorted in descending order by label.
We need more sorting options:
- Sort by Value/Metric used in chart
- Sort by some other metric
- Change sort order, ASC/DESC

**Comment by sfirke** [comment]
Essentially the same functionality described in the first post [here](https://github.com/apache/superset/issues/13792), where a metric like a CASE WHEN could control sort order.  But applied to the Bar Chart v2 with GENERIC_CHART_AXES flag enabled.

**Reply by sfirke** [reply]
I wonder if this PR made this work on the main branch?  https://github.com/apache/superset/pull/21993 I have not tested it but if so, maybe a good candidate for 2.1.0 release.

**Comment by villebro** [comment]
FYI @kamalkeshavani-aiinside @sfirke, this PR adds the option to sort the base axis labels by a sort metric: #23274

**Comment by villebro** [comment]
Slightly related, this PR proposes changing the default sort order of the ECharts charts to equal the legacy NVD3 charts and adds the option to configure the order: #23392

**Reply by sfirke** [reply]
hi @villebro - in the Operational Model Release Strategy group today I mentioned these PRs re: sorting.  If it's quick for you to answer, do you think they would be candidates for a bugfix 2.1.1 release (if they would cherry pick cleanly) or should they be held for the next major release of 2.2 or 3.0?

**Reply by villebro** [reply]
@sfirke I think they should apply fairly cleanly to 2.1.1, as 2.1.0 is fairly fresh, so if we consider these eligible for a patch release I'm sure I can cherry them in.

**Reply by villebro** [reply]
@sfirke I've tagged my sorting PRs for 2.1.1. If you have a second please check if you agree with the PRs I've tagged for inclusion.

**Comment by Vicinius** [comment]
This is really needed 😢 found a workaround on stackoverflow https://stackoverflow.com/questions/73279908/how-to-customize-the-order-of-bars-in-a-superset-chart that case when should help until we have this on... Not the same thing obviously but might be useful for someone reading this thread

**Reply by villebro** [reply]
We've done good progress on this side, and I'd say generic x axis has better sorting now than the legacy charts. If tou're able to try master branch please see if there's still something missing.

**Comment by kangbobi** [comment]
need this
